scripts/mysqld_safe.sh

```diff
@@ -971,9 +988,9 @@ do
 done
 cmd="$cmd $args"
 # Avoid 'nohup: ignoring input' warning
-test -n "$NOHUP_NICENESS" && cmd="$cmd < /dev/null"
+[ -n "$NOHUP_NICENESS" -a $simulate -eq 0 ] && cmd="$cmd < /dev/null"
```

Comment:
I believe you can avoid all further changes if you just do "[ $simulate -eq 0 ] && return" before this line.